Raspi/utils/average.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Calculate a rolling average on a set of data points
+ */
+class RollingAverage {
+  /**
+   * Create a rolling average counter
+   *
+   * @param {number} interval Length of rolling interval window in milliseconds
+   */
+  constructor(interval) {
+    this.interval = interval;
+    this.points = [];
+  }
+
+  /**
+   * Add data point to average
+   *
+   * @param {number} value Value
+   */
+  add(value) {
+    const time = Date.now();
+    this.points.push([time, value]);
+  }
+
+  /**
+   * Calculate the current average value over the rolling interval
+   *
+   * @returns {number} Average value
+   */
+  average() {
+    // Remove all stale values
+    const time = Date.now();
+    this.points = this.points.filter((x) => x[0] >= time - this.interval);
```

Comment:
What are your thoughts on making a fixed length Queue that pushes old data when a new data comes in if the Queue has hit that length. Will be less intensive then filtering an array each time.

Reply:
I think it's a non-issue. We're getting data from the pedals at ~1Hz, and we're sending to the broker at 1Hz. At most the buffer will have 5-10 items. I doubt this will become a bottleneck, but we can fix it if it comes to that.

Reply:
Yup, totally agree. Given that the maximum amount of items in `this.points` will be around ~10 items. Just wanted to raise this point for next time!